storage/client/src/main/java/com/azure/storage/queue/credentials/SASTokenCredential.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.storage.queue.credentials;
+
+import com.azure.core.implementation.util.ImplUtils;
+
+import java.util.HashMap;
+
+/**
+ * Holds a SAS token used for authenticating requests.
+ */
+public final class SASTokenCredential {
+    // Required SAS token pieces
+    private static final String SIGNED_VERSION = "sv";
+    private static final String SIGNED_SERVICES = "ss";
+    private static final String SIGNED_RESOURCE_TYPES = "srt";
+    private static final String SIGNED_PERMISSIONS = "sp";
+    private static final String SIGNED_EXPIRY = "se";
+    private static final String SIGNATURE = "sig";
+
+    // Optional SAS token pieces
+    private static final String SIGNED_START = "st";
+    private static final String SIGNED_PROTOCOL = "spr";
+    private static final String SIGNED_IP = "sip";
+
+    private final String sasToken;
+
+    public SASTokenCredential(String sharedKey) {
+        this.sasToken = sharedKey;
+    }
+
+    public String sasToken() {
+        return sasToken;
+    }
+
+    /**
+     * Creates a SAS token credential from the passed URL query string
+     * @param query URL query used to build the SAS token
+     * @return A SAS token credential if the query param contains all the necessary pieces
+     */
+    public static SASTokenCredential fromQuery(String query) {
+        if (ImplUtils.isNullOrEmpty(query)) {
+            return null;
+        }
+
+        HashMap<String, String> queryParams = new HashMap<>();
+        for (String queryParam : query.split("&")) {
+            String key = queryParam.split("=", 1)[0];
```

Comment:
limit should set to 2 or no limits.
Based on the java doc:
```
If the limit <i>n</i> is greater than zero then the pattern
will be applied at most <i>n</i>&nbsp;-&nbsp;1 times, the array's
length will be no greater than <i>n</i>, and the array's last entry
will contain all input beyond the last matched delimiter.```

The key you get will be the input e.g. "sv=1.0.0"